xbmc/dbwrappers/mysqldataset.cpp

```diff
@@ -941,7 +944,8 @@ void MysqlDatabase::mysqlVXPrintf(
     /* Fetch the info entry for the field */
     infop = &fmtinfo[0];
     xtype = etINVALID;
-    for(idx2=0; idx2<ARRAY_SIZE(fmtinfo); idx2++){
+    for (idx2 = 0; idx2 < fmtinfo.size(); idx2++)
```

Comment:
Here can also a `for (const auto& info : fmInfo)` used, maybe becomes then also `clang` more happy below (strange style there :smirk:) :smile:.

Reply:
updated with a new commit